backend/courses/management/commands/recompute_topics.py

```diff
@@ -151,5 +151,31 @@ def handle(self, *args, **kwargs):
             assert (
                 min_semester in all_semesters()
             ), f"--min-semester={min_semester} is not a valid semester."
-
+        semesters = sorted(
+            [sem for sem in all_semesters() if not min_semester or sem >= min_semester]
+        )
         recompute_topics(min_semester, verbose=True, allow_null_parent_topic=bool(min_semester))
+        if semesters:
+            recompute_historical_probabilities(current_semester=semesters[-1], verbose=True)
+
+
+def recompute_historical_probabilities(current_semester, verbose=False):
+    """
+    Recomputes the historical probabilities for all topics.
+    """
+    if verbose:
+        print("Recomputing historical probabilities for all topics...")
+    topics = Topic.objects.all()
+    length = len(topics)
+    # Iterate over each Topic
+    for i, topic in enumerate(topics):
+        if i % 1000 == 0:
+            print(f"Recomputing topics for semesters >={i}/{length}")
+        # Calculate historical_year_probability for the current topic
+        test = topic.courses.order_by("semester").all()
```

Comment:
Are you sure you can do the order_by before the all? You might be able to I just don't have personal experience with it so want to make sure it's correct.